cpp/ql/test/library-tests/dataflow/fields/complex.cpp

```diff
@@ -41,8 +41,8 @@ void bar(Bar &b)
   // then it tracks that both `a_` and `b_` have followed `f` in _some_ access
   // path somewhere in the search. That makes the library conclude that there
   // could be flow to `b.f.a_` even when the flow was actually to `b.f.b_`.
-  sink(b.f.a()); // flow [FALSE POSITIVE through `b2.f.setB` and `b3.f.setB` in AST, NOT DETECTED by IR]
-  sink(b.f.b()); // flow [FALSE POSITIVE through `b1.f.setA` (AST) and `b3.f.setA` in AST, NOT DETECTED by IR]
+  sink(b.f.a()); //$ast=flow 55:13 $ast=flow 57:13 $f-:ir=flow
+  sink(b.f.b()); //$ast=flow 56:13 $ast=flow 58:13 $f-:ir=flow
```

Comment:
This was previously annotated as a false positive, but I can see that the comment was written before we increased the access-path approximation by one. So perhaps it's no longer a false positive?

Reply:
Yes. On `master` there are 4 sources (and no false positives). If I add another field lookup (i.e., `sink(b.g.f.a())`) then we get the additional false positives the comment above mentions.